src/SubspaceClustering.jl

```diff
@@ -35,34 +54,52 @@ with subspace dimensions `d[1], ..., d[K]`.
 
 # Keyword Arguments
 - `niters::Int=100`: Maximum number of iterations (default is 100).
--'Uinit::Union{Nothing, Vector{AbstractMatrix}}': Initial subspaces for the algorithm. If set to nothing, random subspaces are initialized.
+- `randng::StableRNG=StableRNG(1234)`: Random number generator.
+- 'Uinit::Vector{AbstractMatrix}': A vector of length 'K' containing initial subspace bases. If not provided, they are initialized randomly using randsubspace function.
 
 # Returns
 - 'U::Vector{AbstractMatrix}': A vector of length `K` containing the subspace basis matrices for 'K' clusters.
 - 'c::Vector{Int}': A vector of length `N` containing the cluster assignments for each data point.
 """
-function KSS(X, d; niters=100, Uinit=polar.(randn.(size(X, 1), collect(d))))
+function KSS(X::AbstractMatrix{<:Real},                                                     #in: data matrix with size (D, N)
+			d::Vector{<:Integer};                                                           #in: a vector of subspace dimensions of length K
+			niters::Integer = 100,                                                          #in: number of iterations
+			randng::StableRNG = StableRNG(1234),                                            #in: a random number generator with StableRNG type
+			Uinit::Vector{Matrix{Float64}} = randsubspace(size(X, 1), d; rng=default_rng()) #in: a vector of length K containing initial subspaces
```

Comment:
This should use the `rng` provided by the user. Once the signature for `randsubspace` has been updated as described above, this line should look something like:

```suggestion
			Uinit::AbstractVector{<:AbstractMatrix{<:Real}} = [randsubspace(rng, size(X, 1), dk) for dk in d]) #in: a vector of length K containing initial subspaces
```

I've also updated the type signature here. You mentioned on the phone that you had tried something like `Vector{Matrix{<:Real}}` and it didn't work. To understand more of what's going on here, read about parametric types and `UnionAll` types here:

- https://docs.julialang.org/en/v1/manual/types/#Parametric-Types
- https://docs.julialang.org/en/v1/manual/types/#UnionAll-Types

And try the following experiments!

```julia
Uinit isa Vector{Matrix{<:Real}}
Uinit isa Vector{<:Matrix{<:Real}}
Uinit isa AbstractVector{<:AbstractMatrix{<:Real}}
```

See which are true/false and see if you can explain why!

Reply:
`Uinit isa Vector{Matrix{<:Real}}` - **false** - Because of type invariance, `Vector{Matrix{Float64}}` does not subtype `Vector{Matrix{<:Real}}` as they differ in exact type parameters. `Matrix{Float64}` is an instance within `Matrix{<:Real}`, but not the same as the union type, `Matrix{<:Real}`. 

`Uinit isa Vector{<:Matrix{<:Real}}` - **true** - Since `Pointy{Float64} <: Pointy{<:Real}` is true, we have a `Vector{Matrix{Float64}} <: Vector{<:Matrix{<:Real}` which means a vector whose element type is some subtype of `Matix{<:Real}`

<img width="836" alt="Screenshot 2025-04-04 at 9 59 55 AM" src="https://github.com/user-attachments/assets/3e68839b-fc8b-411f-90a0-308e1d66c96d" />

`Uinit isa AbstractVector{<:AbstractMatrix{<:Real}}` - **true** - Since `Vector{Matrix{Float64}}` is itself a subtype of `AbstractVector{Matrix{Float64}}`, which is a subtype of `AbstractVector{<:AbstractMatrix{<:Real}}`. 

<img width="1185" alt="Screenshot 2025-04-04 at 10 08 41 AM" src="https://github.com/user-attachments/assets/6deabc3d-7499-4682-ac7f-7502ea7429df" />